sweep/bucket_list.go

```diff
@@ -0,0 +1,42 @@
+package sweep
+
+// bucket contains a set of inputs that are not mutually exclusive.
+type bucket pendingInputs
+
+// tryAdd tries to add a new input to this bucket.
+func (b bucket) tryAdd(input *pendingInput) bool {
+	exclusiveGroup := input.params.ExclusiveGroup
+	if exclusiveGroup != nil {
+		for _, input := range b {
+			if input.params.ExclusiveGroup == exclusiveGroup {
```

Comment:
shouldn't this test value equality rather than pointer equality? fwiw the unit test uses the same pointer for all inputs, which explains why this isn't caught

Reply:
good catch. unit test updated too.